agent/api/container/container.go

```diff
@@ -631,9 +640,27 @@ func (c *Container) SetAppliedStatus(status ContainerStatus) bool {
 }
 
 // GetAppliedStatus returns the transitioning status of container
-func (c *Container) GetAppliedStatus() ContainerStatus {
+func (c *Container) GetAppliedStatus() apicontainerstatus.ContainerStatus {
 	c.lock.RLock()
 	defer c.lock.RUnlock()
 
 	return c.AppliedStatus
 }
+
+// ShouldPullWithASMAuth returns true if this container needs to retrieve
+// private registry authentication data from ASM
+func (c *Container) ShouldPullWithASMAuth() bool {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	return c.RegistryAuthentication != nil &&
```

Comment:
I understand this, but would prefer it to be expressed  more explicitly.

Reply:
> would prefer it to be expressed more explicitly.

how do you mean exactly? i dont follow.

Reply:
Try to make it so that it isn't a multi-line boolean expression